### PR TITLE
Refactor "Issue Trigger" for modularity

### LINKED ISSUE #7372: Refactor and consolidate "Issue Trigger" workflow and related files
### Overview
We want our workflows to use consistency in formatting and to make efficient use of modular functions where possible. The "Issue Trigger" workflow can be streamlined by using existing modules.

### Action Items
Open `/add-preliminary-comment/preliminary-update-comment.js`;
- [x] Remove the function `getItemInfo()` and refactor with a call to the already existing `utils/query-issue-info.js` module
- [x] Remove the function `updateItemStatus()` and refactor with a call to the already existing `utils/mutate-issue-status.js` module
- [x] Remove project id, status field id, and status values and refactor with a call to the existing `utils/data/status-field-ids.js` module
- [x] Demonstrate the functionality of "Issue Trigger" after refactoring

### PR DESCRIPTION
**Pull request by t-will-gillis**
Fixes #7372 

### What changes did you make?
  
  - In `preliminary-update-comment.js`, removed functions `getItemInfo()` and `updateItemStatus()` and refactored with calls to existing `utils/query-issue-info.js` and `utils/mutate-issue-status.js`
  - In the same file, removed the constants `PROJECT_ID`, `STATUS_FIELD_ID`, and `statusValues` and refactored with call to existing `utils/_data/status-field-ids.js`

### Why did you make the changes (we will use this info to test)?
  
  - The functions mentioned are already modules so refactored to streamline code

 

### Screenshots of Proposed Changes To The Website (if any, please do not include screenshots of code changes)
- No visual changes
- [Workflow log](https://github.com/t-will-gillis/website/actions/runs/10687314051/job/29624639524) showing successful run

## Notes for Testing
- Refer to [Hack for LA's GitHub Actions (current revision)](https://github.com/hackforla/website/issues/6537#issuecomment-2041147335), especially Tips 6, 7, and 8, if you have not already created you personal environment for testing.
- The files [status-field-ids.js](https://github.com/hackforla/website/blob/gh-pages/github-actions/utils/_data/status-field-ids.js) and [mutate-issue-status.js](https://github.com/hackforla/website/blob/gh-pages/github-actions/utils/mutate-issue-status.js) both are hard-coded with values unique to HfLA. You will need to replace these values with values matching your repo. In order to find these values, go to:
  - [GitHub GraphQL Explorer](https://docs.github.com/en/enterprise-server@3.12/graphql/overview/explorer) 
  - [GitHub GraphQL API](https://github.com/octokit/octokit.js?tab=readme-ov-file#graphql-api-queries) for reference
  - To find your repo's values, first substitute the following into Explorer:  
  ```  
  query findStatusSubfieldIds ($login: String!, $projNum: Int!, $fieldName: String!   ) {
    user(login: $login) {
      projectV2(number: $projNum) {
        id
        field(name:$fieldName) { 
          ... on ProjectV2SingleSelectField { 	
          id options{ 
              id
              name         
              ... on ProjectV2SingleSelectFieldOption {
                id
              }
            } 
          }
        }
      }
    }
  }
  ```  
  and for "Variables":  
  ```  
  {
    "login": "hackforla",
    "projNum": 86,
    "fieldName": "Status"
  }  
  ```  
  but substitute your login name and project number instead. For the project number, click on "Projects" and the number will be listed there.  
- in [get-team-members.js](https://github.com/hackforla/website/blob/gh-pages/github-actions/utils/get-team-members.js) substitute `org: 'hackforla'` on line 16
- When you are ready to test, check that your default branch is set to the name of the testing branch. To change you repo's default branch, click on "Settings", then refer to "Default branch" and if needed select the arrows icon, the branch you want to switch to, and then "Update". Confirm that you understand and want to update the default branch. 
- When you are done testing, be sure to reset your branch to `gh-pages`.
- Now you should be able to test this by creating new issues in your repo and checking that the appropriate comments are added to the issue.

### CODE REVIEW

File: github-actions/trigger-issue/add-preliminary-comment/preliminary-update-comment.js [hunk 1]
```diff
@@ -51,27 +41,32 @@
...
-      await unAssignDev(); // Unassign the developer
-      await addLabel(READY_FOR_DEV_LABEL); // Add 'ready for dev lead' label
+      await unAssignDev();
+      await addLabel(READY_FOR_PRIORITIZATION);
+      console.log(' - remove developer and add `Ready for Prioritization` label');
 
       // Update item's status to "New Issue Approval"
-      const itemInfo = await getItemInfo();
-      await updateItemStatus(itemInfo.id, statusValues.get(New_Issue_Approval));
+      let statusValue = statusFieldIds("New_Issue_Approval")
```
[the remark this answers is not in the document]
Comment (t-will-gillis) [reply github-actions/trigger-issue/add-preliminary-comment/preliminary-update-comment.js]: added semicolon as noted

**Comment by irais-valenzuela** [comment]
Review ETA: 9/13/22
Availability: Wed-Friday

**Comment by t-will-gillis** [comment]
Hi @irais-valenzuela I realize that I should have left more details about testing. I will plan to add info in the next day or so.

**Comment by daras-cu** [comment]
Review ETA: 9/14 EOD
Availability: Weekdays after 6 pm PST, all day Sat/Sun/Mon

**Comment by t-will-gillis** [comment]
Hello @irais-valenzuela and @daras-cu I added testing notes to the PR description. I believe this should get you there but please feel free to message me if you see something. Thanks

**Comment by daras-cu** [comment]
@t-will-gillis I wish I had waited another hour to work on this, I just finished testing without seeing your instructions... At least it was good practice.

**Review by daras-cu — DISMISSED** [review]
@t-will-gillis nice job, you factored out the necessary functions successfully and the formatting all looks good.

I tested in my repo and the action seems to be working as intended:
- The appropriate comment is posted when adding an assignee to an issue
- If the assignee already has an issue, they are removed and the issue is moved back to `New Issue Approval` with the `Ready for Prioritization` label

**Comment by irais-valenzuela** [comment]
Was waiting on testing instructions but will work on today and tommorow.
Review ETA: 9/17/22
Availability: Mon - Thurs

**Comment by t-will-gillis** [comment]
Hi @irais-valenzuela Sounds great- thank you! FYI the two latest commits from today are for resolving a merge conflict and removing duplicate code that resulted in a CodeQL alert...

File: github-actions/trigger-issue/add-preliminary-comment/preliminary-update-comment.js [hunk 2]
```diff
@@ -52,15 +42,29 @@
...
     // Get the latest developer in case there are multiple assignees
     assignee = await getLatestAssignee();
 
     // Check if developer is allowed to work on this issue
     const isAdminOrMerge = await memberOfAdminOrMergeTeam();
-    const isAssignedToAnotherIssues = await assignedToAnotherIssue();
+    const isAssignedToAnotherIssue = await assignedToAnotherIssue();
+
+    // Check if developer is allowed to work on complexity level of the issue
+    const issueComplexityPermitted = await checkComplexityEligibility(
```
[the remark this answers is not in the document]
Comment (t-will-gillis) [reply github-actions/trigger-issue/add-preliminary-comment/preliminary-update-comment.js:54]: This segment of code was inadvertently duplicated- which was causing the 'assignment to declared constant' warning. the duplicated code is removed.

**Comment by irais-valenzuela** [comment]
Hey @t-will-gillis! I've managed to go through the code and understand what these functions are doing and the ask of the ticket but I'm having some trouble with testing. I'm unable to make a copy of the project board. I've tried multiple times to make a copy but keep encountering this issue 'There was a problem copying the project. The workflow "PR added to project" exceeds the maximum of 1 workflows for its type' once I click copy project. I would love some help on this when you get a chance.

**Comment by t-will-gillis** [comment]
Hey @irais-valenzuela argh- okay sorry, you are likely ok. Others have seen that message and you can safely ignore it. (Santiago is writing an issue to fix this in the documents)

**Review by irais-valenzuela — DISMISSED** [review]
Hey @t-will-gillis, nice job on this! The branches are good and the code changes complete ticket goals. I tested this and get the appropriate preliminary comment and multiple issue comment when I pick up an issue or pick one up when I already have one. Also, the appropriate actions trigger following the comments of unassigning the engineer, adding 'ready for prioritization label', and changing issue label to 'new issue approval'. Great job!

**Review by daras-cu — DISMISSED** [review]
@t-will-gillis I did re-test this and everything is still working as intended

**Comment by t-will-gillis** [comment]
Addressed merge conflict

**Pull request merged**